src/core/net/nd6.cpp

```diff
@@ -170,6 +170,16 @@ uint8_t RouteInfoOption::OptionLengthForPrefix(uint8_t aPrefixLength)
     return length;
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+// RaFlagsExtOption
+
+void RaFlagsExtOption::Init(void)
```

Comment:
the Init() method is not called from anywhere - shouldn't there at least be a unit test function that calls it to create an instance and do some simple ops on it?

Reply:
Yes. Not used today. We will use if we start adding it in our emitted RA.

Not sure it is worth adding a unit test for such a simple method.
It will be tested once we include in the emitted RA. (we will parse and check the RA similar to other options).

Reply:
Ok to do this later! Is there already an issue or PR open for adding RA emission with the stub router flag set? For Thread 1.3.1 this is mandatory. Or maybe await the final allocation but that could take a while.

Reply:
Will probably add it in a future PR.  Did not know that it is mandatory for 1.3.1 (thought it was optional).

I agree would be better to have the flag allocated but I understand that it may not be soon (can we face any compatibility issue related to this in future if the allocated bit differs from what we assume it would be).

There may be some aspects to consider. If I remember correctly, there were issues with some devices (running older network stacks) not being able to parse RAs that include this option (they did not ignore it and failed and rejected the whole RA instead?) with a possible workaround of adding the option added at the end?

Reply:
@EskoDijk just submitted https://github.com/openthread/openthread/pull/9486 for this.